<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TrafficControl</class>
 <widget class="QMainWindow" name="TrafficControl">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>690</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Maximum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>trafficControl</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>1181</width>
      <height>661</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="cursor">
     <cursorShape>ArrowCursor</cursorShape>
    </property>
    <property name="currentIndex">
     <number>4</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Sandbox</string>
     </attribute>
     <widget class="QTableView" name="trainListTableView">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>120</y>
        <width>331</width>
        <height>151</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>8</pointsize>
       </font>
      </property>
     </widget>
     <widget class="QPushButton" name="listTrainButton">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>94</width>
        <height>33</height>
       </rect>
      </property>
      <property name="text">
       <string>List Trains</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="tickIntervalSpinBox">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>30</y>
        <width>61</width>
        <height>22</height>
       </rect>
      </property>
      <property name="suffix">
       <string> ms</string>
      </property>
      <property name="minimum">
       <number>20</number>
      </property>
      <property name="maximum">
       <number>1000</number>
      </property>
      <property name="value">
       <number>500</number>
      </property>
     </widget>
     <widget class="QCheckBox" name="runThreadCheckBox">
      <property name="geometry">
       <rect>
        <x>11</x>
        <y>11</y>
        <width>113</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Continue with ticks</string>
      </property>
     </widget>
     <widget class="QLabel" name="threadStatusButton">
      <property name="geometry">
       <rect>
        <x>11</x>
        <y>34</y>
        <width>38</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Interval</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="stepTimeBox">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>420</y>
        <width>49</width>
        <height>33</height>
       </rect>
      </property>
      <property name="suffix">
       <string> s</string>
      </property>
      <property name="maximum">
       <number>3600</number>
      </property>
      <property name="value">
       <number>1</number>
      </property>
     </widget>
     <widget class="QPushButton" name="timeTickButton">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>420</y>
        <width>101</width>
        <height>33</height>
       </rect>
      </property>
      <property name="text">
       <string>Step time</string>
      </property>
     </widget>
     <widget class="QPushButton" name="importNetworkButton">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>470</y>
        <width>171</width>
        <height>33</height>
       </rect>
      </property>
      <property name="text">
       <string>Import Network</string>
      </property>
     </widget>
     <widget class="QPushButton" name="generateNetworkButton">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>510</y>
        <width>201</width>
        <height>33</height>
       </rect>
      </property>
      <property name="text">
       <string>Not used</string>
      </property>
     </widget>
     <widget class="QTableView" name="stationListTableView">
      <property name="geometry">
       <rect>
        <x>870</x>
        <y>40</y>
        <width>300</width>
        <height>581</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>8</pointsize>
       </font>
      </property>
     </widget>
     <widget class="QPushButton" name="listStationButton">
      <property name="geometry">
       <rect>
        <x>870</x>
        <y>10</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>List Stations</string>
      </property>
     </widget>
     <widget class="QTableView" name="trackListTableView">
      <property name="geometry">
       <rect>
        <x>360</x>
        <y>40</y>
        <width>500</width>
        <height>581</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>8</pointsize>
       </font>
      </property>
     </widget>
     <widget class="QPushButton" name="listTrackButton">
      <property name="geometry">
       <rect>
        <x>360</x>
        <y>10</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>ListTracks</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_1">
     <attribute name="title">
      <string>Trains</string>
     </attribute>
     <widget class="QPushButton" name="createTrainButton">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>30</y>
        <width>94</width>
        <height>33</height>
       </rect>
      </property>
      <property name="text">
       <string>Add Train</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="createTrainCapacitySpinBox">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>40</y>
        <width>131</width>
        <height>33</height>
       </rect>
      </property>
      <property name="cursor">
       <cursorShape>IBeamCursor</cursorShape>
      </property>
      <property name="suffix">
       <string> passengers</string>
      </property>
      <property name="maximum">
       <number>600</number>
      </property>
      <property name="value">
       <number>100</number>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>50</y>
        <width>71</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Capacity</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="createTrainNameLineEdit">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>90</y>
        <width>113</width>
        <height>31</height>
       </rect>
      </property>
      <property name="cursor">
       <cursorShape>IBeamCursor</cursorShape>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>100</y>
        <width>41</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Name</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>150</y>
        <width>71</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>ID: </string>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>150</y>
        <width>71</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>TextLabel</string>
      </property>
     </widget>
     <widget class="QFrame" name="frame">
      <property name="geometry">
       <rect>
        <x>390</x>
        <y>10</y>
        <width>251</width>
        <height>181</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QGroupBox" name="groupBox">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>221</width>
         <height>211</height>
        </rect>
       </property>
       <property name="title">
        <string>Speed of train</string>
       </property>
       <widget class="QSpinBox" name="speedSpinBox">
        <property name="geometry">
         <rect>
          <x>130</x>
          <y>30</y>
          <width>91</width>
          <height>33</height>
         </rect>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="suffix">
         <string> km/h</string>
        </property>
        <property name="prefix">
         <string/>
        </property>
        <property name="minimum">
         <number>-180</number>
        </property>
        <property name="maximum">
         <number>180</number>
        </property>
       </widget>
       <widget class="QLCDNumber" name="speedlcdNumber">
        <property name="geometry">
         <rect>
          <x>130</x>
          <y>72</y>
          <width>41</width>
          <height>31</height>
         </rect>
        </property>
       </widget>
       <widget class="QPushButton" name="emergencyBrakePushButton">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>120</y>
          <width>171</width>
          <height>33</height>
         </rect>
        </property>
        <property name="text">
         <string>Emergency Brake</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_5">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>40</y>
          <width>111</width>
          <height>23</height>
         </rect>
        </property>
        <property name="text">
         <string>Desired Speed</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_6">
        <property name="geometry">
         <rect>
          <x>180</x>
          <y>80</y>
          <width>41</width>
          <height>23</height>
         </rect>
        </property>
        <property name="text">
         <string>km/h</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_7">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>80</y>
          <width>101</width>
          <height>23</height>
         </rect>
        </property>
        <property name="text">
         <string>Actual Speed</string>
        </property>
       </widget>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Stations</string>
     </attribute>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>Tracks</string>
     </attribute>
    </widget>
    <widget class="QWidget" name="tab_map">
     <attribute name="title">
      <string>Map</string>
     </attribute>
     <widget class="QGraphicsView" name="mapGraphicsView">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>16</width>
        <height>51</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="graphicsInitButton">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>0</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Init Scene</string>
      </property>
     </widget>
     <widget class="QQuickWidget" name="mapQuickWidget">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>39</y>
        <width>1171</width>
        <height>601</height>
       </rect>
      </property>
      <property name="resizeMode">
       <enum>QQuickWidget::SizeRootObjectToView</enum>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuTrafficControl">
    <property name="title">
     <string>TrafficControl</string>
    </property>
   </widget>
   <addaction name="menuTrafficControl"/>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QQuickWidget</class>
   <extends>QWidget</extends>
   <header>QtQuickWidgets/QQuickWidget</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
